libraries/AP_Notify/Buzzer.cpp

```diff
@@ -68,15 +68,57 @@ void Buzzer::update()
 
 void Buzzer::update_pattern_to_play()
 {
+    if (!_flags.initialize_done) {
+        // play a quick beep at startup
+        if (!_flags.startup_beep) {
+            _flags.startup_beep = true;
+            play_pattern(SINGLE_BUZZ);
+            return;
+        }
+        // play beeps when system initialization complete
+        if (!_flags.initialize_started) {
```

Comment:
This flag does nothing.

Suggest changing this to:

```
        if (!_flags.initialise_buzz_done && !AP_Notify::flags.initialising) {
```

Reply:
I found that 'AP_Notify::flags.initialising' is false early in the startup, goes true during the initialization, and then goes false again.  So, the tracking via flags is needed to prevent the beeps playing before the initialization has begun.